src/Sarif/CodeGenHints.json

```diff
@@ -1,156 +1,166 @@
+      "kind": "DictionaryHint"
     }
   ],
   "AnnotatedCodeLocation.Properties": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint" 
     }
   ],
   "file": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.ClassNameHint, Microsoft.Json.Schema.ToDotNet",
-      "className": "FileData"
+      "kind": "ClassNameHint",
+      "arguments": {
+        "className": "FileData"
+      }
     }
   ],
   "FileData.Properties": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "Location.Properties": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "Result.Properties": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "Rule.MessageFormats": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "Rule.Properties": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "Run.Files": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "Run.LogicalLocations": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "Run.Properties": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint" 
     }
   ],
   "Run.Rules": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.DictionaryHint, Microsoft.Json.Schema.ToDotNet"
+      "kind": "DictionaryHint"
     }
   ],
   "rule": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.InterfaceHint, Microsoft.Json.Schema.ToDotNet",
-      "description": "Interface exposed by objects that provide information about analysis rules."
+      "kind": "InterfaceHint",
+      "arguments": {
+        "description": "Interface exposed by objects that provide information about analysis rules."
+      }
     }
   ],
   "SarifLog.Version": [
     {
-      "$type": "Microsoft.Json.Schema.ToDotNet.EnumHint, Microsoft.Json.Schema.ToDotNet",
-      "typeName": "SarifVersion",
-      "description": "Possible values for the SARIF schema version.",
-      "zeroValue": "Unknown",
-      "enum": [
-        "OneZeroZeroBetaThree"
-      ]
+      "kind": "EnumHint",
+      "arguments": {
+        "typeName": "SarifVersion",
+        "description": "Possible values for the SARIF schema version.",
+        "zeroValue": "Unknown",
+        "enumValues": [
```

Comment:
Oh, and also, I changed the name of the JSchema property `EnumHint.Enum` to `EnumHint.EnumValues`.